Userland/Services/Taskbar/main.cpp

```diff
@@ -226,15 +226,29 @@ ErrorOr<NonnullRefPtr<GUI::Menu>> build_system_menu(WindowRefence& window_ref)
     g_themes = Gfx::list_installed_system_themes();
     auto current_theme_name = GUI::ConnectionToWindowServer::the().get_system_theme();
 
+    auto& current_theme = g_themes[0];
+    for (auto& theme : g_themes) {
+        if (theme.name == current_theme_name)
+            current_theme = theme;
+    }
+
     {
         int theme_identifier = 0;
         for (auto& theme : g_themes) {
-            auto action = GUI::Action::create_checkable(theme.name, [theme_identifier](auto&) {
+            auto action = GUI::Action::create_checkable(theme.name, [&current_theme, theme_identifier](auto&) {
                 auto& theme = g_themes[theme_identifier];
                 dbgln("Theme switched to {} at path {}", theme.name, theme.path);
                 auto success = GUI::ConnectionToWindowServer::the().set_system_theme(theme.path, theme.name, false);
                 VERIFY(success);
+                current_theme = theme;
```

Comment:
This assumes that `Taskbar` is the authority on the system theme, but it could also change in other ways. It would be better to listen for some kind of theme update notification here.